File: JVM/jnetcall-java-tests/src/test/java/jnetcall/java/tests/io/TcpTransportTest.java
package jnetcall.java.tests.io;

import java.util.concurrent.atomic.AtomicInteger;

import org.javatuples.Pair;

import jnetcall.java.api.io.ISendTransport;
import jnetcall.java.impl.io.net.TcpTransport;

public final class TcpTransportTest extends TransportTest {

    private static AtomicInteger _offset = new AtomicInteger();

    private static int getNextOffset() {
        return _offset.incrementAndGet();
    }

    @Override
    protected Pair<ISendTransport, ISendTransport> getBoth() {
        int offset = getNextOffset();
        int portOne = 12001 + offset;
        int portTwo = 12051 + offset;
        TcpTransport left = new TcpTransport(
                Encoding,
                "localhost", portOne,
                "localhost", portTwo
        );
        TcpTransport right = new TcpTransport(
                Encoding,
                "localhost", portTwo,
                "localhost", portOne
        );
        return Pair.with(left, right);
    }
}

File: NET/JNetCall.Sharp.Tests/IO/TcpTransportTest.cs
using System.Threading;
using JNetCall.Sharp.API.IO;
using JNetCall.Sharp.Impl.IO.Net;

namespace JNetCall.Sharp.Tests.IO
{
    public sealed class TcpTransportTest : TransportTest
    {
        private static int _offset;
        private static int NextOffset => Interlocked.Increment(ref _offset);

        protected override (ISendTransport, ISendTransport) GetBoth()
        {
            var offset = NextOffset;
            var portOne = 12001 + offset;
            var portTwo = 12051 + offset;
            var left = new TcpTransport(
                Encoding,
                "localhost", portOne,
                "localhost", portTwo
            );
            var right = new TcpTransport(
                Encoding,
                "localhost", portTwo,
                "localhost", portOne
            );
            return (left, right);
        }
    }
}